pkg/prometheus/promcfg.go

```diff
@@ -907,7 +910,8 @@ func (cg *ConfigGenerator) generateProbeConfig(
 	}
 	labeler := namespacelabeler.New(cpf.EnforcedNamespaceLabel, cpf.ExcludedFromEnforcement, false)
 
-	if m.Spec.Targets.StaticConfig != nil {
+	switch {
```

Comment:
(nit) emphasize why the order is important?
```suggestion
	// As stated in the CRD documentation, if both StaticConfig and Ingress are defined, the former takes precedence which is why the first case statement checks for m.Spec.Targets.StaticConfig.
	switch {
```